packages/customer/src/models/customer-group-customer.ts

```diff
@@ -0,0 +1,65 @@
+import { DAL } from "@medusajs/types"
+import { generateEntityId } from "@medusajs/utils"
+import {
+  BeforeCreate,
+  ManyToOne,
+  Entity,
+  OnInit,
+  OptionalProps,
+  PrimaryKey,
+  Property,
+} from "@mikro-orm/core"
+import Customer from "./customer"
+import CustomerGroup from "./customer-group"
+
+type OptionalGroupProps = DAL.EntityDateColumns // TODO: To be revisited when more clear
+
+@Entity({ tableName: "customer_group_customer" })
+export default class CustomerGroupCustomer {
+  [OptionalProps]: OptionalGroupProps
+
+  @PrimaryKey({ columnType: "text" })
+  id!: string
+
+  @ManyToOne({
+    entity: () => Customer,
+    fieldName: "customer__id",
+    index: "IDX_customer_group_customer_customer_id",
+  })
+  customer: Customer
+
+  @ManyToOne({
+    entity: () => CustomerGroup,
+    fieldName: "customer_group_id",
+    index: "IDX_customer_group_customer_group_id",
+  })
+  customer_group: CustomerGroup
+
+  @Property({ columnType: "jsonb", nullable: true })
+  metadata?: Record<string, unknown> | null
```

Comment:
```suggestion
  metadata: Record<string, unknown> | null = null
```

Reply:
What about created_at,updated_at ?

Reply:
they always have a default value and they are part of the optional props so they are handled a bit differently